cms/djangoapps/contentstore/helpers.py

```diff
@@ -286,6 +287,27 @@ class StaticFileNotices:
     error_files: list[str] = Factory(list)
 
 
+def _rewrite_static_asset_references(downstream_xblock: XBlock, substitutions: dict[str, str], user_id: int) -> None:
+    """
+    Rewrite the static asset references in the OLX string to point to the new locations in the course.
+    """
+    store = modulestore()
+    if hasattr(downstream_xblock, "data"):
+        data_with_substitutions = downstream_xblock.data
+        for old_static_ref, new_static_ref in substitutions.items():
+            data_with_substitutions = _replace_strings(
+                data_with_substitutions,
+                old_static_ref,
+                new_static_ref,
+            )
+        downstream_xblock.data = data_with_substitutions
+        if store is not None:
+            store.update_item(downstream_xblock, user_id)
```

Comment:
when would `store` be `None`? `modulestore()` always returns the modulestore singleton as far as I know.

I think a better approach is something like "if this block is using a modulestore, update it":

```suggestion
        downstream_xblock.save()  # This works in the new XBlock runtime but is a no-op in modulestores
        if hasattr(downstream_xblock.runtime, "modulestore"):
            # Do this to actually save the changes in modulestore-based runtime:
            downstream_xblock.runtime.modulestore.update_item(downstream_xblock, user_id)
```

Reply:
> when would store be None? modulestore() always returns the modulestore singleton as far as I know.

That's right! I'm removing the check here: 8b3660e21495a78b5d53b0126e766bd22539e6e8
Do you think we need any other checks here?

Reply:
I would recommend something like I suggested above, but technically we don't need it for now - it would only be important when we start converting courses to Learning Core.